internal/venice-common/src/main/java/com/linkedin/venice/pubsub/api/PubSubPosition.java

```diff
@@ -7,6 +7,82 @@
  * Represents a position of a message in a partition of a topic.
  */
 public interface PubSubPosition {
+  /**
+   * A special position representing the earliest available message in a partition. All pub-sub adapters must support
+   * this position, and all pub-sub client implementations should interpret it as the earliest retrievable message in
+   * the partition. Implementations must map this position to the corresponding earliest offset or equivalent marker
+   * in the underlying pub-sub system.
+   */
+  PubSubPosition EARLIEST = new PubSubPosition() {
+    @Override
+    public int comparePosition(PubSubPosition other) {
+      throw new IllegalStateException("Cannot compare EARLIEST position");
+    }
+
+    @Override
+    public long diff(PubSubPosition other) {
+      throw new IllegalStateException("Cannot diff EARLIEST position");
+    }
+
+    @Override
+    public PubSubPositionWireFormat getPositionWireFormat() {
+      throw new IllegalStateException("Cannot serialize EARLIEST position");
+    }
+
+    @Override
+    public String toString() {
+      return "EARLIEST";
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj == this;
+    }
+
+    @Override
+    public int hashCode() {
+      return 1;
```

Comment:
Maybe we should use negative values for magic positions?

Reply:
1 looks strange though, btw who is expected to call this method? If no one should we set something negative or even throw exception.

Reply:
It's usually called when we use these objects (as keys) in hash based collections

Reply:
I'll update it to use a negative value to indicate it's a special/magic value. It might not be ideal, but we typically use negative values for internal purposes.